dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScope.kt

```diff
@@ -217,7 +222,25 @@ internal open class RumViewScope(
         val startedKey = keyRef.get()
         val shouldStop = (event.key == startedKey) || (startedKey == null)
         if (shouldStop && !stopped) {
-            GlobalRum.updateRumContext(getRumContext().copy(viewType = RumViewType.NONE))
+            GlobalRum.updateRumContext(
+                getRumContext().copy(
+                    viewType = RumViewType.NONE,
+                    viewId = null,
+                    viewName = null,
+                    viewUrl = null,
+                    actionId = null
+                ),
+                applyOnlyIf = { currentContext ->
+                    if (currentContext.viewId == this.viewId) {
```

Comment:
When is this possible ?

Reply:
In the current code it is not possible, because if another view starts before current one is stopped, it will be stopped receiving `StartView` for another view (so next `StopView` for this view won't have an effect), but it is just a safeguard to prevent possible bugs if code is changed and allows to have such situation, because accessing `GlobalRum` is dangerous (and so that we are aware of such situation).

Reply:
Yup...this is why I asked because I could not find this possible with the current logic. Anyway, is good to have the safeguard.